src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -747,5 +777,88 @@ void ClusterTaskManager::RemoveFromBacklogTracker(const Task &task) {
   }
 }
 
+void ClusterTaskManager::FreeLocalTaskResources(std::shared_ptr<WorkerInterface> worker) {
+  cluster_resource_scheduler_->FreeLocalTaskResources(worker->GetAllocatedInstances());
+  worker->ClearAllocatedInstances();
+  cluster_resource_scheduler_->FreeLocalTaskResources(
+      worker->GetLifetimeAllocatedInstances());
+  worker->ClearLifetimeAllocatedInstances();
+}
+
+bool ClusterTaskManager::ReleaseCpuResourcesAndMarkWorkerAsBlocked(
```

Comment:
> Can you add unit tests for the two methods related to releasing CPU resources/marking workers as blocked? It would be good to check:
> 
> * calling this twice only releases resources/marks the worker as blocked once (and same for ReturnCpuResourcesAndMarkWorkerAsUnblocked)
> * resources are the same before and after releasing/returning resources
> * only CPU resources are released/returned.

Okey, I will do it.